classwork/classwork2.2/main.cpp

```diff
@@ -0,0 +1,150 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <cstdlib>
+#include <print>
+
+// Function to split string by a delimiter
+std::vector<std::string> split(const std::string &str, char delimiter) {
+    std::vector<std::string> tokens;
+    std::string token;
+    std::istringstream tokenStream(str);
+    while (std::getline(tokenStream, token, delimiter)) {
+        tokens.push_back(token);
+    }
+    return tokens;
+}
+
+// Function to validate if a string represents a valid integer in the range [0, 255]
+bool isValidColorValue(const std::string &str) {
+    try {
+        int value = std::stoi(str);
+        return (value >= 0 && value <= 255);
+    } catch (...) {
+        return false;
+    }
+}
+
+// Function to validate if an integer is in the range [0, 255]
+bool isValidColorValue(int value) {
+    return (value >= 0 && value <= 255);
+}
+
+// Function to parse a line into a vector of pixels
+std::vector<std::tuple<int, int, int>> parseLine(const std::string &line) {
+    std::vector<std::tuple<int, int, int>> pixels;
+    std::vector<std::string> parts = split(line, ' ');
+    for (const std::string &part : parts) {
+        std::vector<std::string> colorValues = split(part, ',');
+        if (colorValues.size() != 3 ||
+            !isValidColorValue(colorValues[0]) ||
+            !isValidColorValue(colorValues[1]) ||
+            !isValidColorValue(colorValues[2])) {
+            throw std::invalid_argument("Invalid pixel format");
+        }
+        pixels.emplace_back(std::stoi(colorValues[0]), std::stoi(colorValues[1]), std::stoi(colorValues[2]));
+    }
+    return pixels;
+}
+
+// Function to process the image based on the favorite color
+void processImage(const std::string &inputFileName, const std::tuple<int, int, int> &favoriteColor, const std::string &outputFileName, const std::tuple<int, int, int> &unfavoriteColor = std::make_tuple(-1, -1, -1)) {
```

Comment:
This function is too long and it is definitely has more then one responsibility.
The signature of the function is to complicated and not clear for reader.
Let's introduce Color or Pixel class or structure with methods to validate data and so on.. Then we can introduce Image class that will encapsulate the data from line 60. As a result you should get Image class with API to load, transform and save image to the file.